pallets/compliance-manager/src/lib.rs

```diff
@@ -732,6 +766,34 @@ impl<T: Trait> Module<T> {
         }
         asset_rules_with_results
     }
+
+    fn verify_rules_complexity(
+        asset_rules: &Vec<AssetTransferRule>,
+        default_issuer_count: usize,
+    ) -> DispatchResult {
+        let mut complexity = 0usize;
+        for asset_rule in asset_rules {
+            for rule in asset_rule
+                .sender_rules
+                .iter()
+                .chain(asset_rule.receiver_rules.iter())
+            {
+                let (claims, issuers) = rule.complexity();
+                if issuers == 0 {
```

Comment:
you can use `match` and avoid using mutable variable.

something like that 

```
let complexity = match issuers {
0 => complexity.saturating_add(claims.saturating_mul(default_issuer_count));
_ => complexity.saturating_add(claims.saturating_mul(issuers));
};
```

Reply:
Mutability is not required because of the if/else. In fact, you can use a non-mutuable variable in an if-else as well. Mutability is required to initialize the variable as 0. In case there are no rules/empty rules, the complexity falls back to 0.